lib/faraday/adapter/http_gem.rb

```diff
@@ -21,10 +29,12 @@ def call(env)
               :headers => env[:request_headers],
               :proxy => proxy_options(env)
             }.merge(socket_options(env)))
-        rescue HTTP::ConnectionError => e
-          raise Faraday::Error::ConnectionFailed, e
-        rescue OpenSSL::SSL::SSLError => e
-          raise Faraday::SSLError, e
+        rescue *HTTP_EXCEPTIONS => err
+          if defined?(OpenSSL) && OpenSSL::SSL::SSLError === err
```

Comment:
Fair enough!

On Thu, Mar 31, 2016 at 10:05 AM, bbozo <notifications@github.com> wrote:

> In lib/faraday/adapter/http_gem.rb
> <https://github.com/lostisland/faraday/pull/564#discussion_r58058624>:
>
> > @@ -21,10 +29,12 @@ def call(env)
> >                :headers => env[:request_headers],
> >                :proxy => proxy_options(env)
> >              }.merge(socket_options(env)))
> > -        rescue HTTP::ConnectionError => e
> > -          raise Faraday::Error::ConnectionFailed, e
> > -        rescue OpenSSL::SSL::SSLError => e
> > -          raise Faraday::SSLError, e
> > +        rescue *HTTP_EXCEPTIONS => err
> > +          if defined?(OpenSSL) && OpenSSL::SSL::SSLError === err
>
> I shamelessly took the exception handling pattern from the Net::HTTP
> adapter implementation
> https://github.com/lostisland/faraday/blob/master/lib/faraday/adapter/net_http.rb
> (reasoning it's one of the oldest and most mature) and OpenSSL isn't a
> part of the HTTP gem's gemspec so I guess there are scenarios when this
> would be an issue.
>
> —
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub
> <https://github.com/lostisland/faraday/pull/564/files/352359388abbd0168eb68df00a3174b74e099812..1d23259c74ade0fb1d3edc5daf2a7627f2e89e71#r58058624>
>